### PR TITLE
Text area using variables

### PR DESCRIPTION
**Pull request by thomtrp**
- Adding multiline props to component
- Update design and logic accordingly 
- Fix hotkey scope for right drawer

https://github.com/user-attachments/assets/65ff9641-71a4-4828-a62b-e09327b63150

### CODE REVIEW

**Review by martmull — APPROVED** [review]
LGTM

File: packages/twenty-front/src/modules/workflow/search-variables/components/VariableTagInput.tsx [hunk 1]
```diff
@@ -130,6 +168,29 @@ export const VariableTagInput = ({
...
+            state.schema.nodes.hardBreak.create(),
+          );
+
+          view.dispatch(transaction);
+
+          return true;
+        }
+        return false;
+      },
+    },
```
Comment (Devessier) [inline packages/twenty-front/src/modules/workflow/search-variables/components/VariableTagInput.tsx:190]: Why do we implement that logic? It seems that Tiptap should handle it, shouldn't it?

File: packages/twenty-front/src/modules/workflow/search-variables/components/VariableTagInput.tsx [hunk 2]
```diff
@@ -130,6 +168,29 @@ export const VariableTagInput = ({
...
+    enableInputRules: false,
+    enablePasteRules: false,
```
Comment (Devessier) [inline packages/twenty-front/src/modules/workflow/search-variables/components/VariableTagInput.tsx:192]: Does it have something to do with the fact that we can't copy and paste from other `<VariableTagInput />` components and get the variables correctly wired? I'm not saying that we absolutely want that behavior, I'm just wondering what problem you solved by disabling `enabledPasteRules`.


https://github.com/user-attachments/assets/61857248-b93f-42d6-bc28-c3b84c2a60f6

**Review by Devessier — APPROVED** [review]
Great work!! I left a few questions to better understand the challenges you faced! Thank you.

File: packages/twenty-front/src/modules/workflow/search-variables/components/VariableTagInput.tsx (hunk 1 above)
Comment (thomtrp) [reply packages/twenty-front/src/modules/workflow/search-variables/components/VariableTagInput.tsx:190]: Tip tap does not perform hardbreak on `Enter`. The behavior is not great by default

File: packages/twenty-front/src/modules/workflow/search-variables/components/VariableTagInput.tsx (hunk 2 above)
Comment (thomtrp) [reply packages/twenty-front/src/modules/workflow/search-variables/components/VariableTagInput.tsx:192]: I am not sure. Tiptap by default enable a lot of features that we do not need. Following the doc, their recommendation is for exemple to use a `StarterKit` for several extensions. And enabling input and paste rules that I did not fully understand https://tiptap.dev/docs/editor/extensions/custom-extensions/extend-existing#paste-rules. By default I prefer removing these and put back only the ones I need later.

But to answer your video, no, the behavior si the same with or without those paste rules

File: packages/twenty-front/src/modules/workflow/search-variables/components/VariableTagInput.tsx (hunk 1 above)
Comment (Devessier) [reply packages/twenty-front/src/modules/workflow/search-variables/components/VariableTagInput.tsx:190]: Thanks for the info!

File: packages/twenty-front/src/modules/workflow/search-variables/components/VariableTagInput.tsx (hunk 2 above)
Comment (Devessier) [reply packages/twenty-front/src/modules/workflow/search-variables/components/VariableTagInput.tsx:192]: Okay! Great! I agree with you 💯

**Pull request merged**